### ISSUE TITLE
[BUG]: AzureCLI@2 az login fails when using certificate for authentication starting from version 2.67.0 of Azure CLI

### ISSUE THREAD

**Issue by jan-msc** [microsoft/azure-pipelines-tasks#20672]
Starting from version 2.67.0, for `az login`, `--password` no longer accepts service principal certificate. Use `--certificate` to pass a service principal certificate.
AzureCLI@2 does not use the `--certificate` parameter to pass the service principal certificate resulting in:
`ERROR: AADSTS7000215: Invalid client secret provided. Ensure the secret being sent in the request is the client secret value, not the client secret ID, for a secret added to app '***'.
The error may be caused by passing a service principal certificate with --password. Please note that --password no longer accepts a service principal certificate. To pass a service principal certificate, use --certificate instead.`

**Comment by NakagawaMakoto** [comment]
I am experiencing the same issue. Is there any way to downgrade azure cli to prior version?

**Comment by jan-msc** [comment]
@NakagawaMakoto : Azure CLI has no downgrade functionality but you can uninstall it and install the desired version (2.66.0) instead.

**Comment by NakagawaMakoto** [comment]
Yes, of course, it is me who is responsible for the version of azcli on the pipeline agent's host. I should have known better. Thank you for reminding me of the fact.

**Comment by RichieBzzzt** [comment]
To work around this issue we added this step to our builds. Would love to see the issue fixed though!

```- script: |
    apt-get remove -y azure-cli
    apt-get install -y azure-cli=2.66.0-1~focal```

**Comment by mhever** [comment]
This is breaking user pipelines on our side, this should be fixed asap.

**Comment by robkendrick** [comment]
> As a workaround, adding this task should help on windows host
> 
>  - powershell: |
>     wmic product where "name='Microsoft Azure CLI (64-bit)'" call uninstall /nointeractive
>     Invoke-WebRequest -Uri https://azcliprod.blob.core.windows.net/msi/azure-cli-2.66.0-x64.msi -OutFile .\AzureCLI.msi; 
>     Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet';
>   displayName: Downgrade azure CLI

Thanks for this--I wasn't having much luck with the `msiexec.exe /x` uninstaller! Unfortunately, it takes the Hosted Agents anywhere from six to fourteen minutes to complete the uninstall.

In the interim, I've added Federated Credentials to the App Registration in Entra ID, then associated that with a new Service Connection in AzDO, and that works with no issues--pipelines are back to being speedy. The great part about this method (thus far) is that because it's the same AppID it uses the same Service Principal (Enterprise Apps in EID), so you don't have to go update the RBAC for every resource.

For those who haven't done this before, here's a quick rundown of how to do it:
1. Go to your Service Connections under Project Settings (can also just use the standard URI: `https://dev.azure.com/<orgName>/<projectName | projectID>/_settings/adminservices`)
2. Click \[ New Service Connection \] -> Azure Resource Manager -> \[ Next \] -> Workload Identity federation (manual) -> \[ Next \].
3. Give the Service Connection a name that is distinctive from the existing version of the service principal.
   - I would advise naming it something that indicates they're related (e.g.: original is "ServiceConnection-Dev-Certificate", make this one "ServiceConnection-Dev-Federated"). Tick/enable "Grant access permission to all pipelines" only if needed, then click \[ Next \].
5. Leave that tab as-is because you'll need the provided `Issuer` and `Subject Identifier` values. Open a new tab, then go to AAD/EID.
6. Go to your App Registration for the existing Service Connection (or a new one if you decide to go that route), then click on the Manage -> Certificates & secrets Blade.
7. Click Federated Credentials, then `+ Add credential`.![Image](https://github.com/user-attachments/assets/e3cb6f77-a808-4a0d-98ef-282c05d11c6d)
8. Enter/Select the following:
   a. **Federated credential scenario**: Other
   b. **Issuer**: Enter the Issuer from the AzDO Service Connection in the previous tab.
   c. **Subject Identifier**: Enter this value from the AzDO Service Connection.
   d. Give the credential a name.
   e. Click \[ Add \].
9. Click the **Overview** blade on the left and then copy/save/notate the "Application (client) ID" for this App.
10. Go back to the AzDO tab and finish setting up the Service Connection--remaining instructions assume a scope of "Subscription."
   a. **Service Principal ID** is the AppID copied from Step 9.
   b. **Subscription ID** and **Name** can be found easily on the [Subscriptions blade](https://portal.azure.com/#view/Microsoft_Azure_Billing/SubscriptionsBladeV2).
   c. **Tenant ID** can be quickly pulled from the [directory listing](https://portal.azure.com/#settings/directory).
11. Click \[ Verify and Save \].
12. Update your pipelines with the new Service Connection name and test.

**Comment by KKing299** [comment]
@robkendrick - What you said is partially a workaround. It doesn't work out for MSPs since one app registration can only have 20/30 federated credential. Whenever you create a service connection it will take up one federated credential. Let's say if someone wants to manage multiple tenants then it wouldn't work because of restriction on the number of federated credential an app registration can have.
@v-schhabra Does MS have any updates? This needs to be fixed ASAP! This is breaking production setup

**Comment by v-schhabra** [comment]
Hi @KKing299 
The issue has been fixed in this PR https://github.com/microsoft/azure-pipelines-tasks/pull/20698
Please let us know if you are having any issues.

**Comment by KKing299** [comment]
Can someone look at this as well? https://github.com/microsoft/azure-pipelines-tasks/issues/20722
Similar situation for ARM template task.


Keshav Khanna

FIS | Empowering the Financial World [cid:4b724b59-23f7-4070-ac1b-e74650a6c425] <https://www.facebook.com/FIStoday> [cid:f04f1fa4-d400-499e-9b99-28b22167c2ef] <https://twitter.com/FISGlobal> [cid:9740d7d1-2361-4b80-aff9-f16432353350] <https://www.linkedin.com/company/fis>


________________________________

**Comment by v-schhabra** [comment]
Hi @KKing299 
please share the latest updates.

**Comment by v-schhabra** [comment]
Hi @jan-msc @RichieBzzzt @NakagawaMakoto @robkendrick 
Please update here if still you have encountered this issue else will close it.

**Comment by jan-msc** [comment]
Hi @v-schhabra , it's working for us now.

**Comment by AlexandraRamirezMS** [comment]
@v-schhabra can you please review this other item?
same issue when using the ARM Template deployment task https://github.com/microsoft/azure-pipelines-tasks/issues/20722

**Comment by KKing299** [comment]
@v-schhabra Can someone please look at https://github.com/microsoft/azure-pipelines-tasks/issues/20722 ?
We need some traction please.

**Comment by claudiusandu** [comment]
any updates about the fix? will upgrading the version will fix the issue (i am on 2.51 version currently). As far as I understand either wait until end of February or update the CLI to 2.66.

**Comment by claudiusandu** [comment]
any updates about the fix? will upgrading the version will fix the issue (i am on 2.51 version currently). As far as I understand either wait until end of February or update the CLI to 2.66.

**Comment by v-schhabra** [comment]
Hi @claudiusandu 
I would prefer you to update the CLI to 2.66.0
